st4/FileDecoder.py

```diff
@@ -0,0 +1,59 @@
+from LSP.plugin.core.protocol import Error
+from LSP.plugin.core.registry import LspTextCommand
+from LSP.plugin.core.typing import Any
+from LSP.plugin.core.url import filename_to_uri  # TODO: deprecated in a future version
+
+import os
+import sublime
+
+class FileDecoderCommand(LspTextCommand):
+    session_name = "metals"
+
+    _jvm_extentions = {'java', 'scala', 'class'}
+    _tasty_extentions = {'scala', 'tasty'}
+    _scala_extentions = {'scala', 'tasty'}
+    _sematicdb_extentions = {'scala', 'java', 'semanticdb'}
+    _decoders = {
+            'javap': _jvm_extentions,
+            'javap-verbose': _jvm_extentions,
+            'cfr': _jvm_extentions,
+            'tasty-decoded': _tasty_extentions,
+            'semanticdb-compact': _sematicdb_extentions,
+            'semanticdb-detailed': _sematicdb_extentions,
+            'semanticdb-proto': _sematicdb_extentions
+        }
+
+    def is_enabled(self, decoding_type: str) -> bool:
+        if super().is_enabled(None, None):
+            extension = os.path.splitext(self.view.file_name())[1][1:]
+            accepted_extentions = self._decoders.get(decoding_type)
+            return extension in accepted_extentions
+        else:
+            return False
+
+    def run(self, edit: sublime.Edit, decoding_type: str) -> None:
+        uri = 'metalsDecode:' + filename_to_uri(self.view.file_name()) + '.' + decoding_type
+        session = self.session_by_name(self.session_name)
+        if session:
+            params = {
+                "command": 'file-decode',
+                "arguments": [uri]
+            }
+
+            def handle_response(response: Any) -> None:
+                command_name = 'file-decode'
+                if isinstance(response, Error) or 'error' in response:
```

Comment:
Normally `if isinstance(response, Error)` is enough but the command's response doesn't respect the spec of `workspace/executeCommand`.

I think 'error' shouldn't be included into   
https://github.com/scalameta/metals/blob/3f211b03d6c2d1ff4d5fa936d50a4aab013100fb/metals/src/main/scala/scala/meta/internal/metals/FileDecoderProvider.scala#L39 
But rather in separate error response, see LSP [spec](https://microsoft.github.io/language-server-protocol/specification#workspace_executeCommand)

```
Response:

result: any | null
error: code and message set in case an exception happens during the request.
```